course03/miguel/project/asteroidmanager.cc

```diff
@@ -0,0 +1,55 @@
+#include "asteroidmanager.h"
+#include "utils.h"
+
+namespace SpaceImpactGame {
+
+AsteroidManager::AsteroidManager(const vec2& bounds, int numAsteroids)
```

Comment:
In general, any class named "XXXXManager" is a code smell. The problem with such class is that its responsibility is not well defined. What is a "manager" supposed to do? It will end up doing a lot / too many things. 

Here it seems like you just want to have a Scene class or something. A class that paints / updates objects, without owning them.